custom/enikshay/management/commands/mark_successfull_99dots_records.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import absolute_import, print_function
+
+import csv
+from datetime import datetime
+
+from django.core.management.base import BaseCommand
+
+from corehq.apps.hqcase.utils import update_case
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.motech.repeaters.dbaccessors import (get_repeat_record_count,
+                                                 iter_repeat_records_by_domain)
+from corehq.motech.repeaters.models import RepeatRecord
+from corehq.util.log import with_progress_bar
+from custom.enikshay.case_utils import get_person_case, get_adherence_cases_from_episode
+
+
+class Command(BaseCommand):
+    """Finds all records that were originally marked as failed due to a 99DOTS API
+    error, but then were subsequently marked as successful. Sends all other
+    pertinent records for these cases to 99DOTS.
+
+    """
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument('--commit', action='store_true')
+
+    def handle(self, domain, **options):
+        # For all successful registration records
+        # If any have an attempt that id "A patient with this beneficiary_id already exists"
+        # Check the episode case. If this doesn't have "dots_99_registered" then set this property to "true"
+        self.commit = options['commit']
+        repeater_id = 'dc73c3da43d42acd964d80b287926833'  # 99dots register
+        accessor = CaseAccessors(domain)
+        existing_message = "A patient with this beneficiary_id already exists"
+        count = get_repeat_record_count(domain, repeater_id, state="SUCCESS")
+        records = iter_repeat_records_by_domain(domain, repeater_id, state="SUCCESS")
+
+        cases_to_update = set()
+        print("Filtering successful cases")
+        for repeat_record in with_progress_bar(records, length=count):
+            if repeat_record.attempts and any(
```

Comment:
Whats the use of checking `repeat_record.attempts` first here? I guess `any` would return false in case there are no attempts.